Introduce the new changelog tool (#18726)
ckeditor/ckeditor5: docs/framework/contributing/changelog-entries.md

      </td>
    </tr>
    <tr>
      <td><code>scope</code></td>
      <td>❌ No</td>
      <td>Affected package(s), using short names like <code>ckeditor5-core</code>.</td>
    </tr>
    <tr>
      <td><code>closes</code></td>
      <td>❌ No</td>
      <td>List of issues this change resolves. Use numbers (<code>123</code>), full references (<code>ckeditor/ckeditor5#123</code>), or full URLs.</td>
    </tr>
    <tr>
      <td><code>see</code></td>
      <td>❌ No</td>
      <td>Related issues that provide context but are not directly resolved by this change. Same format as <code>closes</code>.</td>
    </tr>
    <tr>
      <td><code>communityCredits</code></td>
      <td>❌ No</td>
      <td>GitHub usernames of external contributors who should be credited for this change.</td>
    </tr>
    <tr>
      <td><em>(body)</em></td>
      <td>✅ Yes</td>
      <td>After the frontmatter, add a short and meaningful summary of the change. Optionally include extended context or rationale.</td>
    </tr>
  </tbody>
</table>

<info-box>
    **Tip**: Keep the summary clear and user-facing &ndash; this is what will appear in the final changelog.
</info-box>

> The changelog entry format is designed to be both human-friendly and machine-readable. It uses a simple frontmatter structure followed by a short description of the change. Each field in the frontmatter serves a specific purpose, from determining the entry's visibility to linking it with related issues or acknowledging community contributions.

Author: ```suggestion
The changelog entry format is designed to be both human-friendly and machine-readable. It uses a simple frontmatter structure followed by a short description of the change. Each field in the frontmatter serves a specific purpose, from determining the visibility of the entry to linking it with related issues or acknowledging community contributions.
```

      <th style="width: 10%;">Release</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Feature</td>
      <td><code>minor</code></td>
      <td>A new feature. Introduces user-facing functionality.</td>
    </tr>
    <tr>
      <td>Fix</td>
      <td><code>patch</code></td>
      <td>A bug fix. Use also for small improvements that do not qualify as new features.</td>
    </tr>
    <tr>
      <td>Other</td>
      <td><code>patch</code></td>
      <td>Enhancement or refactor. It's not a fix or feature. Example: public API cleanup.</td>
    </tr>
    <tr>
      <td>Major breaking change</td>
      <td><code>major</code></td>
      <td>A change in the integration layer or the plugin development API. See {@link updating/versioning-policy versioning policy} for details.</td>
    </tr>
    <tr>
      <td>Minor breaking change</td>
      <td><code>minor</code></td>
      <td>Low-level customizable API layer. See {@link updating/versioning-policy versioning policy} for details.</td>
    </tr>
  </tbody>
</table>

### Package name (`scope`)

Changes affect one or more packages. List the package that was most impacted by the change first.

> However, it is possible to skip this part if many packages are affected. This usually indicates a generic change. In this case, having all the packages listed would reduce the changelog's readability.

Author: ```suggestion
However, it is possible to skip this part if many packages are affected. This usually indicates a generic change. In this case, having all the packages listed would reduce the readability of the changelog.
```

The package name is based on the npm package name, but the `@ckeditor/` prefix is stripped.

If your change is related to the main package, use `ckeditor5` as the package name.

<info-box>
    If the commit introduces a breaking change across the entire project (a generic change), you do not need to specify the package name.
</info-box>

### Referencing issues

When creating PRs that address specific issues, use the following messages to indicate them.

* `Closes` &ndash; When the PR resolves an issue.
* `See` &ndash; When the PR references an issue but has not resolved it yet.

Both fields (`closes` and `see`) can contain multiple references, but they must follow the same format:

* `14724` &ndash; A simple issue number.
* `ckeditor/ckeditor5#14724` &ndash; A full reference to an issue in the CKEditor 5 repository.
* `https://github.com/ckeditor/ckeditor5/issues/14724` &ndash; A full URL to an issue in the CKEditor 5 repository.

### Giving credit

> When closing a R submitted by a non-core contributor, add information about the contributor to the changelog entry file using the `communityCredits` field. It should contain a list of GitHub usernames of contributors who should be credited for this change.

Author: ```suggestion
When closing a PR submitted by a non-core contributor, add information about the contributor to the changelog entry file using the `communityCredits` field. It should contain a list of GitHub usernames of contributors who should be credited for this change.
```

* `https://github.com/ckeditor/ckeditor5/issues/14724` &ndash; A full URL to an issue in the CKEditor 5 repository.

### Giving credit

When closing a non-core contributor's PR, add information about the contributor to the changelog entry file using the `communityCredits` field. It should contain a list of GitHub usernames of contributors who should be credited for this change.

### Description

Write a concise and meaningful summary of the change. This main message will appear in the public changelog, so keep it clear, user-facing, and relevant.

Use the `ClassName#methodName()` format when referencing methods. This ensures consistency across all entries.

**Example:**

```
MarkerCollection#has()
```

You may include multiple sentences if additional context is helpful.

### Examples of correct entry formatting

<info-box>
    Unlike the previous Git-based system, which captured all commit types, including internal changes, the new file-based changelog focuses exclusively on public, user-facing changes, ensuring the final changelog remains clear and relevant to end users.
</info-box>

A new feature without any breaking changes.

```md
---
type: Feature
scope:
- ckeditor5-ui

>   closes:

Author: Fixed in b1bf4f4.

- 1
---

Added support for RTL languages.

RTL content will now be rendered correctly.
```

A generic bug fix for an existing feature that affects many packages (closes two tickets):

```md
---
type: Fix
closes:
   - 2
   - 3
---

The editor will be great again.
```

An improvement that is not backward compatible and sent by a non-core contributor. Public API was changed:

```md
---
type: Other
scope:
- ckeditor5-utils
closes:
- 9
---

Extracted the `utils#foo()` to a separate package.
```

```md
---
type: Feature
scope:
    - ckeditor5-engine
closes:

>     - 9

Reviewer: In these examples, there are sometimes 4 spaces, sometimes 3 spaces, sometimes no spaces.

Author: Fixed in b1bf4f4.
